pkg/snet/packet_conn.go

```diff
@@ -115,7 +115,7 @@ type SCIONPacketConnMetrics struct {
 // packets.
 type SCIONPacketConn struct {
 	// Conn is the connection to send/receive serialized packets on.
-	Conn *net.UDPConn
+	Conn UDPConn
```

Comment:
This change isn't likely to break clients but it could still happen and it is therefore a breaking change. If we're okay with breaking clients, I think we should have a discussion on whether we really want to provide an interface here that offers the entire functionality of net.UDPConn. I'm not even sure what the motivation for exporting the bare net.UDPConn is in the first place. Once we have better understanding of this, we could perhaps also provide a more restricted interface or even extend the type snet.PacketConn with that functionality.